### PR TITLE
Replace scikit GradientBoostingClassifier with XGBClassifier

### PR DESCRIPTION
**Pull request by tcfuji**
Also includes a test and dependency in `setup.py`.

### CODE REVIEW

**Comment by tcfuji** [comment]
@rhiever I'm not too familiar with travis. Should I just add `pip install xgboost==$XGBOOST_VERSION` to `.travis_install`?

**Comment by rhiever** [comment]
I believe you need to do the following. @rasbt can confirm.

* [ ] Add the required version of XGBoost to the lines around https://github.com/rhiever/tpot/blob/master/.travis.yml#L7

* [ ] pip install XGBoost similar to how DEAP is installed here: https://github.com/rhiever/tpot/blob/master/ci/.travis_install.sh#L46

* [ ] print the XGBoost version similar to how the others are printed here: https://github.com/rhiever/tpot/blob/master/ci/.travis_install.sh#L58

* [ ] Also print the XGBoost version similarly here: https://github.com/rhiever/tpot/blob/master/ci/.travis_test.sh#L17

**Comment by rasbt** [comment]
I am not using boosting and don't know if XGBoost has any other dependencies. However, if it can be installed via `pip install xgboost` then the steps Randy outlined above are the way to go. Maybe set up a fresh virtual environment with numpy, scipy, and scikit-installed and test if `pip install xgboost` is sufficient or if you need something like `pip install -r requirements.txt`

**Comment by rhiever** [comment]
Thanks @rasbt! You should give XGBoost a try. Even without OpenMP, it seems to be streets ahead of regular gradient boosting.

**Comment by rasbt** [comment]
You are welcome! I dunno, someday maybe, but I feel like XGBoost is more of a ML competition / Kaggle thing :P . Currently, I am more excited about toying around with TensorFlow :)

**Comment by rhiever** [comment]
To each their own. If you look at the examples above, XGBoost is very easy to use. Same interface as sklearn models. I hope the sklearn team will integrate XGBoost soon.

**Comment by rhiever** [comment]
@tcfuji: Will review this tomorrow. Looks good from a high-level glance.

**Comment by rhiever** [comment]
Looks good. Thanks again @tcfuji!

**Comment by rhiever** [comment]
@tcfuji, when you get a moment, can you please submit a separate PR to update the docs? Currently, they say we still use the [sklearn GBC](http://rhiever.github.io/tpot/reference/pipeline_operators/models/classifiers/ensemble/GradientBoostingClassifier/). We need to update that.

The docs are located [here](https://github.com/rhiever/tpot/tree/master/docs/sources).

You can view the docs as you change them by running `mkdocs serve` in the `docs` directory. (This will require you to `pip install mkdocs` if you haven't already.)

**Comment by rhiever** [comment]
I think we should simply remove the sklearn GBC entry and make a new entry for XGBoost. You can likely use the sklearn GBC entry as a template so the changes will be minimal.
